bot/common/guild_select.py

```diff
@@ -18,24 +18,24 @@
 from bot.config import Redis
 
 
-async def get_thread(user_id, key):
+async def get_thread(user_id, key, cache=None):
     val = json.loads(key)
     thread = val.get("thread")
     step = val.get("step")
     message_id = val.get("message_id")
     guild_id = val.get("guild_id")
     if thread == ThreadKeys.ONBOARDING.value:
-        return await Onboarding(user_id, step, message_id, guild_id)
+        return await Onboarding(user_id, step, message_id, guild_id, cache)
     elif thread == ThreadKeys.UPDATE_PROFILE.value:
-        return await UpdateProfile(user_id, step, message_id, guild_id)
+        return await UpdateProfile(user_id, step, message_id, guild_id, cache)
     elif thread == ThreadKeys.INITIAL_CONTRIBUTIONS.value:
-        return await InitialContributions(user_id, step, message_id, guild_id)
+        return await InitialContributions(user_id, step, message_id, guild_id, cache)
     elif thread == ThreadKeys.GUILD_SELECT.value:
-        return await GuildSelect(user_id, step, message_id, guild_id)
+        return await GuildSelect(user_id, step, message_id, guild_id, cache)
     elif thread == ThreadKeys.REPORT.value:
-        return await Report(user_id, step, message_id, guild_id)
+        return await Report(user_id, step, message_id, guild_id, cache)
     elif thread == ThreadKeys.POINTS.value:
-        return await Points(user_id, step, message_id, guild_id)
+        return await Points(user_id, step, message_id, guild_id, cache)
```

Comment:
Are these changes being used? I don't see the cache being passed anywhere to get thread

Reply:
It's being passed in OverrideThredStep.send. needed it for the DM flow.